contrib/lib/src/json.rs

```diff
@@ -138,7 +141,14 @@ impl<'a, T: Deserialize<'a>> FromData<'a> for Json<T> {
     fn transform(r: &Request, d: Data) -> Transform<Outcome<Self::Owned, Self::Error>> {
         let size_limit = r.limits().get("json").unwrap_or(LIMIT);
         let mut s = String::with_capacity(512);
-        match d.open().take(size_limit).read_to_string(&mut s) {
+        match d.open().take(size_limit + 1).read_to_string(&mut s) {
+            Ok(_) if s.len() as u64 == size_limit + 1 => {
```

Comment:
Let's do `if s.len() as u64 > size_limit` - that should make refactoring less brittle.

On that note, the number of bytes read is already returned by `read_to_string` -- `Ok(n) if n as u64 > size_limit` should work as well.

Reply:
Oh, yeah, of course :laughing:

I updated the PR.